lib/addons/fluxcd/index.ts

```diff
@@ -0,0 +1,96 @@
+// lib/fluxcd_addon.ts
+import { Construct } from 'constructs';
+import merge from "ts-deepmerge";
+import { ClusterInfo, Values } from "../../spi";
+import { createNamespace } from "../../utils";
+import { HelmAddOn, HelmAddOnProps, HelmAddOnUserProps } from "../helm-addon";
+import { FluxGitRepository } from "./gitrepository";
+import * as spi from "../../spi";
+import { KubernetesManifest } from 'aws-cdk-lib/aws-eks/lib/k8s-manifest';
+/**
+ * User provided options for the Helm Chart
+ */
+export interface FluxCDAddOnProps extends HelmAddOnUserProps {
+  /**
+   * To Create Namespace using CDK
+   */    
+  createNamespace?: boolean;
+
+  /**
+   * Optional values for `GitRepository` Source to produce an Artifact for a Git repository revision.
+   */
+  bootstrapRepo?: spi.ApplicationRepository;
+}
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+const defaultProps: HelmAddOnProps & FluxCDAddOnProps = {
+  name: "fluxcd-addon",
+  namespace: "flux-system",
+  chart: "flux2",
+  version: "2.7.0",
+  release: "blueprints-fluxcd-addon",
+  repository: "https://fluxcd-community.github.io/helm-charts",
+  values: {},
+  createNamespace: true
+};
+
+/**
+ * Main class to instantiate the Helm chart
+ */
+export class FluxCDAddOn extends HelmAddOn {
+
+  readonly options: FluxCDAddOnProps;
+
+  constructor(props?: FluxCDAddOnProps) {
+    super({...defaultProps, ...props});
+    this.options = this.props as FluxCDAddOnProps;
+  }
+
+  deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+    const cluster = clusterInfo.cluster;
+    let values: Values = populateValues(this.options);
+    values = merge(values, this.props.values ?? {});
+
+    if( this.options.createNamespace == true){

```

Comment:
There is quite a bit of code duplication in if/else statement. It is all done because of namespace creation. Please use the pattern that we established in the ack addon [here](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/lib/addons/ack/index.ts#L77). 
statements like `addHelmChart`, `createGitRepository` and such should only appear once.